sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -250,5 +265,22 @@ class QueryExecution(val sparkSession: SparkSession, val logical: LogicalPlan) {
     def codegenToSeq(): Seq[(String, String)] = {
       org.apache.spark.sql.execution.debug.codegenStringSeq(executedPlan)
     }
+
+    /**
+     * Dumps debug information about query execution into the specified file.
+     */
+    def toFile(path: String): Unit = {
+      val filePath = new Path(path)
+      val fs = FileSystem.get(filePath.toUri, sparkSession.sessionState.newHadoopConf())
+      val writer = new OutputStreamWriter(fs.create(filePath))
```

Comment:
cc @zsxwing Could you help review this function?

Reply:
`val writer = new BufferedWriter(new OutputStreamWriter(fs.create(filePath), UTF_8))`